### PR TITLE
langchain_cohere: add cohere as a partner package

### PR DESCRIPTION
**Pull request by billytrend-cohere**
Description: adds support for langchain_cohere

### CODE REVIEW

File: libs/partners/cohere/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+repository = "https://github.com/langchain-ai/langchain"
+license = "MIT"
+
+[tool.poetry.urls]
+"Source Code" = "https://github.com/langchain-ai/langchain/tree/master/libs/partners/cohere"
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+langchain-core = ">=0.0.12"
+cohere = "5.0.0a12"
```
Comment (billytrend-cohere) [inline libs/partners/cohere/pyproject.toml:16]: TODO: update to 5.0.0 today

File: libs/partners/cohere/langchain_cohere/__init__.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from langchain_cohere.chat_models import ChatCohere
+from langchain_cohere.embeddings import CohereEmbeddings
+from langchain_cohere.llms import BaseCohere
+from langchain_cohere.rag_retrievers import CohereRagRetriever
+
+__all__ = [
+    "BaseCohere",
```
Comment (baskaryan) [inline libs/partners/cohere/langchain_cohere/__init__.py:7]: probably don't need/want to export BaseCohere here? not something for end users i assume

File: libs/partners/cohere/tests/integration_tests/test_llms.py
Comment (baskaryan) [inline libs/partners/cohere/tests/integration_tests/test_llms.py:1]: uncomment or delete

**Review by baskaryan — COMMENTED** [review]
overall lgtm! should update imports in the docs as well

**Comment by baskaryan** [comment]
> overall lgtm! should update imports in the docs as well

as in in `docs/docs/integrations/chat_models/cohere.ipynb` etc

File: docs/docs/integrations/chat/cohere.ipynb
Comment (baskaryan) [inline docs/docs/integrations/chat/cohere.ipynb:1]: you'll want to update the pip instal command at the top of each doc as well

Comment (harry-cohere) [reply docs/docs/integrations/chat/cohere.ipynb:1]: I've done this for the notebooks that had pip install - please can you take another look @baskaryan ? thank you

**Review by efriis — APPROVED** [review]
Going to merge and release this as a `0.1.0rc0` to unblock cohere team, and would love to get the readme filled out before publishing a 0.1.0!

Would also love to remove the LLM from the partner package (just leave in community) if you're planning on deprecating (legacy) those endpoints

https://docs.cohere.com/docs/migrating-from-cogenerate-to-cochat

**Pull request merged**